include/yboard.h

```diff
@@ -102,16 +103,66 @@ class YBoardV3 {
      * representing the name of the sound file to play. The return type is a boolean
      * value (true or false). True corresponds to the sound being played
      * successfully, and false corresponds to an error playing the sound. The sound
-     * file must be stored on the microSD card. After this function is called, the
-     * loop_speaker function must be called often to playback the sound on the speaker.
+     * file must be stored on the microSD card.
      */
     bool play_song_from_sd(const char *filename);
 
+    /* This is similar to the function above, except that it will start the song playing
+     * in the background and return immediately. The song will continue to play in the
+     * background until it is stopped with the stop_audio function, another song is
+     * played, play_notes is called, or the song finishes. After this function is called, the
+     * loop_speaker function must be called often to playback the sound on the speaker.
+     */
+    bool play_song_from_sd_background(const char *filename);
+
     /*
      *  This function sets the speaker volume. The volume is an integer between 0
-     * and 100. A volume of 0 is off, and a volume of 100 is full volume.
+     * and 10. A volume of 0 is off, and a volume of 10 is full volume.
+     */
+    void set_sd_song_volume(uint8_t volume);
```

Comment:
Does this just set the volume when playing a song from the SD card? Or does it control the volume of the speaker regardless if it is playing a song from the SD card or playing notes? If it is the latter, I would suggest we name it `set_speaker_volume`. If it is the former, then how do I control the volume when I am playing notes?

Reply:
Volume ended up being a bit tricker than I thought.  The issue is that both the tones and the wave files are populated into a fairly large buffer before being asynchronously send out to I2C via another thread.  Once the data is in the buffer, you area committed on the volume.  While I suppose we could modify it in place before sending to the I2S, I worried that would add more processing overhead.

For the audio tones, the library syntax I followed allows for changing the volume as part of the note sequence string, which eliminates this issue, since the volume will be pre-determined as the buffer is built up.

This is why there is now a separate function for volume of wave files.  It still will have a slight delay due to buffering.  Right now the code fills the entire buffer (~6 seconds -- make large to handle longest notes), but now that I think about it, I think I can just restrict the wave to use less of the buffer to reduce this audio delay.  I'll make that change.